cisc/cisc.go

```diff
@@ -874,3 +874,299 @@ func followUpdate(c *cli.Context) error {
+		return err
+	}
+
+	if !isCert(cert) {
+		return errors.New("Please give a certificate")
+	}
+	log.Info("Verify the validity of the cert:")
+	if !isValid(cert) {
+		return errors.New("Certificate is not valid, can't add it to proposal storage ")
+	}
+
+	id, err := cfg.findSC(c.Args().Get(2))
+	if err != nil {
+		return err
+	}
+	if id == nil {
+		scList(c)
+		return errors.New("Please give skipchain-id")
+	}
+
+	prop := id.GetProposed()
+	log.Info("Valid Certificate, added to proposal storage")
+	prop.Storage[domain] = cert
+	cfg.proposeSendVoteUpdate(id, prop)
+	return cfg.saveConfig(c)
+}
+
+// Verify the validity of a certificate by giving as argument the key
+// corresponding to this latter
+func certVerify(c *cli.Context) error {
+	if c.NArg() < 1 {
+		return errors.New("Please give the certificate key for verification")
+	}
+	cfg := loadConfigOrFail(c)
+	id, err := cfg.findSC(c.Args().Get(1))
+	if err != nil {
+		return err
+	}
+	if id == nil {
+		scList(c)
+		return errors.New("Please give skipchain-id")
+	}
+
+	k := c.Args().Get(0)
+
+	cert := id.Data.Storage[k]
+
+	if !isCert(cert) {
+		return errors.New("The values do not correspond to a certificate")
+	}
+	log.Info("Verify the validity of the cert:")
+	if !isValid(cert) {
+		return errors.New("Certificate is not valid")
+	}
+	return cfg.saveConfig(c)
+}
+
+// Renew a certificate by giving the domain name/key
+func certRenew(c *cli.Context) error {
+	cfg := loadConfigOrFail(c)
+	if c.NArg() < 1 {
+		return errors.New("Please give a domain name")
+	}
+
+	domain := c.Args().Get(0)
+	id, err := cfg.findSC(c.Args().Get(1))
+	if err != nil {
+		return err
+	}
+	if id == nil {
+		scList(c)
+		return errors.New("Please give skipchain-id")
+	}
+
+	log.Info("Checking the certificate")
+
+	if _, ok := id.Data.Storage[domain]; !ok {
+		return errors.New("Didn't find key " + domain + " in the config")
+	}
+
+	cert := id.Data.Storage[domain]
+	if !isCert(cert) {
+		return errors.New("The values do not correspond to a certificate")
+	}
+
+	// Renew the cert (see certificate.go)
+	newcert, err := renewCert(cert)
+	if err != nil {
+		return errors.New("Error while renewing certificate: " + err.Error())
+	}
+
+	// Check the certificate
+	log.Info("Verify the validity of the cert:")
+	if !isValid(newcert) {
+		return errors.New("Certificate is not valid, can't add it to proposal storage ")
+	}
+	prop := id.GetProposed()
+	log.Info("Valid Certificate, added to proposal storage")
+	prop.Storage[domain] = newcert
+	cfg.proposeSendVoteUpdate(id, prop)
+	return cfg.saveConfig(c)
+}
+
+// Revoke a certificate by giving the key corresponding to the certificate and
+// the register key of this certificate. This certificate will be then deleted
+// from the skipchain
+func certRevoke(c *cli.Context) error {
+	cfg := loadConfigOrFail(c)
+	if c.NArg() < 2 {
+		return errors.New("Please give the certificate to delete and the register key of the certificate")
+	}
+
+	id, err := cfg.findSC(c.Args().Get(2))
```

Comment:
Can you please add a method `cfg.findSCOrList` that includes the `if id == nil` part?